IPython/html/nbextensions.py

```diff
@@ -173,75 +175,59 @@ def install_nbextension(files, overwrite=False, symlink=False, user=False, prefi
     # make sure nbextensions dir exists
     ensure_dir_exists(nbext)
     
-    if isinstance(files, string_types):
-        # one file given, turn it into a list
-        files = [files]
-    if isinstance(files, (list,tuple)):
-        # list given, turn into dict
-        _files = {}
-        for path in map(cast_unicode_py2, files):
-            if path.startswith(('https://', 'http://')):
-                destination = urlparse(path).path.split('/')[-1]
-            elif path.endswith('.zip') or _safe_is_tarfile(path):
-                destination = str(uuid.uuid4()) # ignored for archives
-            else:
-                destination = basename(path)
-            _files[destination] = path
-        files = _files
+    if isinstance(path, (list, tuple)):
+        raise TypeError("path must be a string pointing to a single extension to install; call this function multiple times to install multiple extensions")
     
-    for dest_basename,path in (map(cast_unicode_py2, item) for item in files.items()):
```

Comment:
The previous code called `cast_unicode_py2` on both the path and the destination name; it looks like the new code only calls it on the path.